PreferencesView/Sources/PreferencesView/PressCommandBuilder.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Foundation
+
+@resultBuilder
+public enum PressCommandsBuilder {
```

Comment:
Since we now have two of these similar array-building result builders, let's instead make a generic `ArrayBuilder<Content>` and typealias this `PressCommandsBuilder` and `KeyCommandsBuilder`.

```swift

typealias KeyCommandsBuilder = ArrayBuilder<KeyCommandAction>
typealias PressCommandsBuilder = ArrayBuilder<PressCommandAction>

@resultBuilder
public enum ArrayBuilder<Content> {

    public static func buildBlock(_ components: [Content]...) -> [Content] {
        components.flatMap { $0 }
    }

    public static func buildExpression(_ expression: Content) -> [Content] {
        [expression]
    }

    public static func buildOptional(_ component: [Content]?) -> [Content] {
        component ?? []
    }

    public static func buildEither(first component: [Content]) -> [Content] {
        component
    }

    public static func buildEither(second component: [Content]) -> [Content] {
        component
    }

    public static func buildArray(_ components: [[Content]]) -> [Content] {
        components.flatMap { $0 }
    }
}
```